connectors/webservice/translator-ws/src/main/java/org/teiid/translator/ws/BinaryWSProcedureExecution.java

```diff
@@ -146,7 +142,13 @@ public void execute() throws TranslatorException {
                 ds = new InputStreamFactory.BlobInputStreamFactory((Blob)payload);
             }
 
-            this.returnValue = dispatch.invoke(ds);
+            ClassLoader originalCCL = Thread.currentThread().getContextClassLoader();
```

Comment:
Was this to fix the SOAP errors in the log you were seeing? Just curious.

Reply:
This change was to fix an error seen in the server log when accessing a WebService through TEIID:
12 Jan 2022 06:09:47,614	[Worker12_QueryProcessorQueue2391]	WARNING	SpringBusFactory-	Initial attempt to create application context was unsuccessful.: org.springframework.beans.factory.BeanDefinitionStoreException: IOException parsing XML document from class path resource [META-INF/cxf/cxf.xml]; nested exception is java.io.FileNotFoundException: class path resource [META-INF/cxf/cxf.xml] cannot be opened because it does not exist
	at org.springframework.spring@5.1.14.RELEASE//org.springframework.beans.factory.xml.XmlBeanDefinitionReader.loadBeanDefinitions(XmlBeanDefinitionReader.java:345)
	at org.apache.cxf@3.3.6//org.apache.cxf.bus.spring.ControlledValidationXmlBeanDefinitionReader.internalLoadBeanDefinitions(ControlledValidationXmlBeanDefinitionReader.java:152)
	at org.apache.cxf@3.3.6//org.apache.cxf.bus.spring.ControlledValidationXmlBeanDefinitionReader.access$000(ControlledValidationXmlBeanDefinitionReader.java:66)
	at org.apache.cxf@3.3.6//org.apache.cxf.bus.spring.ControlledValidationXmlBeanDefinitionReader$1.run(ControlledValidationXmlBeanDefinitionReader.java:139)
	at org.apache.cxf@3.3.6//org.apache.cxf.bus.spring.ControlledValidationXmlBeanDefinitionReader$1.run(ControlledValidationXmlBeanDefinitionReader.java:137)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at org.apache.cxf@3.3.6//org.apache.cxf.bus.spring.ControlledValidationXmlBeanDefinitionReader.loadBeanDefinitions(ControlledValidationXmlBeanDefinitionReader.java:137)
	at org.springframework.spring@5.1.14.RELEASE//org.springframework.beans.factory.xml.XmlBeanDefinitionReader.loadBeanDefinitions(XmlBeanDefinitionReader.java:305)
	at org.springframework.spring@5.1.14.RELEASE//org.springframework.beans.factory.support.AbstractBeanDefinitionReader.loadBeanDefinitions(AbstractBeanDefinitionReader.java:188)
	at org.springframework.spring@5.1.14.RELEASE//org.springframework.context.support.AbstractXmlApplicationContext.loadBeanDefinitions(AbstractXmlApplicationContext.java:124)
	at org.apache.cxf@3.3.6//org.apache.cxf.bus.spring.BusApplicationContext.loadBeanDefinitions(BusApplicationContext.java:335)
	at org.springframework.spring@5.1.14.RELEASE//org.springframework.context.support.AbstractRefreshableApplicationContext.refreshBeanFactory(AbstractRefreshableApplicationContext.java:133)
	at org.springframework.spring@5.1.14.RELEASE//org.springframework.context.support.AbstractApplicationContext.obtainFreshBeanFactory(AbstractApplicationContext.java:636)
	at org.springframework.spring@5.1.14.RELEASE//org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:521)
	at org.apache.cxf@3.3.6//org.apache.cxf.bus.spring.BusApplicationContext$1.run(BusApplicationContext.java:107)
	at org.apache.cxf@3.3.6//org.apache.cxf.bus.spring.BusApplicationContext$1.run(BusApplicationContext.java:105)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at org.apache.cxf@3.3.6//org.apache.cxf.bus.spring.BusApplicationContext.<init>(BusApplicationContext.java:105)
	at org.apache.cxf@3.3.6//org.apache.cxf.bus.spring.SpringBusFactory.createApplicationContext(SpringBusFactory.java:157)
	at org.apache.cxf@3.3.6//org.apache.cxf.bus.spring.SpringBusFactory.createBus(SpringBusFactory.java:148)
	at org.apache.cxf@3.3.6//org.apache.cxf.bus.spring.SpringBusFactory.createBus(SpringBusFactory.java:124)
	at org.apache.cxf@3.3.6//org.apache.cxf.bus.spring.SpringBusFactory.createBus(SpringBusFactory.java:94)
	at org.jboss.teiid.resource-adapter.webservice@16.0.0//org.teiid.ws.cxf.BaseWSConnection$HttpDispatch.getBus(BaseWSConnection.java:142)
	at org.jboss.teiid.resource-adapter.webservice@16.0.0//org.teiid.ws.cxf.BaseWSConnection$HttpDispatch.invoke(BaseWSConnection.java:158)
	at org.jboss.teiid.resource-adapter.webservice@16.0.0//org.teiid.ws.cxf.BaseWSConnection$HttpDispatch.invoke(BaseWSConnection.java:118)
	at org.jboss.teiid.translator.ws@16.0.0//org.teiid.translator.ws.BinaryWSProcedureExecution.execute(BinaryWSProcedureExecution.java:149)
	at org.jboss.teiid@16.0.0//org.teiid.dqp.internal.datamgr.ConnectorWorkItem.execute(ConnectorWorkItem.java:405)
	at jdk.internal.reflect.GeneratedMethodAccessor151.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.jboss.teiid@16.0.0//org.teiid.dqp.internal.datamgr.ConnectorManager$1.invoke(ConnectorManager.java:228)
	at org.jboss.teiid@16.0.0//com.sun.proxy.$Proxy171.execute(Unknown Source)
	at org.jboss.teiid@16.0.0//org.teiid.dqp.internal.process.DataTierTupleSource.getResults(DataTierTupleSource.java:302)
	at org.jboss.teiid@16.0.0//org.teiid.dqp.internal.process.DataTierTupleSource$1.call(DataTierTupleSource.java:108)
	at org.jboss.teiid@16.0.0//org.teiid.dqp.internal.process.DataTierTupleSource$1.call(DataTierTupleSource.java:104)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at org.jboss.teiid@16.0.0//org.teiid.dqp.internal.process.FutureWork.run(FutureWork.java:59)
	at org.jboss.teiid@16.0.0//org.teiid.dqp.internal.process.DQPWorkContext.runInContext(DQPWorkContext.java:283)
	at org.jboss.teiid@16.0.0//org.teiid.dqp.internal.process.ThreadReuseExecutor$RunnableWrapper.run(ThreadReuseExecutor.java:124)
	at org.jboss.teiid@16.0.0//org.teiid.dqp.internal.process.ThreadReuseExecutor$2.run(ThreadReuseExecutor.java:212)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.io.FileNotFoundException: class path resource [META-INF/cxf/cxf.xml] cannot be opened because it does not exist
	at org.springframework.spring@5.1.14.RELEASE//org.springframework.core.io.ClassPathResource.getInputStream(ClassPathResource.java:180)
	at org.springframework.spring@5.1.14.RELEASE//org.springframework.beans.factory.xml.XmlBeanDefinitionReader.loadBeanDefinitions(XmlBeanDefinitionReader.java:331)
	... 42 more